src/peripheral/mod.rs

```diff
@@ -646,3 +931,122 @@ pub struct Tpiu {
     /// TPIU Type
     pub _type: RO<u32>,
 }
+
+/// Cache and branch predictor maintenance operations register block
+#[repr(C)]
+#[cfg(armv7m)]
+pub struct Cbp {
+    /// I-cache invalidate all to PoU
+    pub iciallu: WO<u32>,
+    reserved0: RW<u32>,
+    /// I-cache invalidate by MVA to PoU
+    pub icimvau: WO<u32>,
+    /// D-cache invalidate by MVA to PoC
+    pub dcimvac: WO<u32>,
+    /// D-cache invalidate by set-way
+    pub dcisw: WO<u32>,
+    /// D-cache clean by MVA to PoU
+    pub dccmvau: WO<u32>,
+    /// D-cache clean by MVA to PoC
+    pub dccmvac: WO<u32>,
+    /// D-cache clean by set-way
+    pub dccsw: WO<u32>,
+    /// D-cache clean and invalidate by MVA to PoC
+    pub dccimvac: WO<u32>,
+    /// D-cache clean and invalidate by set-way
+    pub dccisw: WO<u32>,
+    /// Branch predictor invalidate all
+    pub bpiall: WO<u32>,
+}
+
+#[cfg(armv7m)]
+const CBP_SW_WAY_POS: u32 = 30;
+#[cfg(armv7m)]
+const CBP_SW_WAY_MASK: u32 = 0x3 << CBP_SW_WAY_POS;
+#[cfg(armv7m)]
+const CBP_SW_SET_POS: u32 = 5;
+#[cfg(armv7m)]
+const CBP_SW_SET_MASK: u32 = 0x1FF << CBP_SW_SET_POS;
+
+#[cfg(armv7m)]
+impl Cbp {
+    /// I-cache invalidate all to PoU
+    #[inline(always)]
+    pub fn iciallu(&self) {
+        unsafe { self.iciallu.write(0); }
+    }
+
+    /// I-cache invalidate by MVA to PoU
+    #[inline(always)]
+    pub fn icimvau(&self, mva: u32) {
+        unsafe { self.icimvau.write(mva); }
```

Comment:
Is the whole u32 range a valid input here? What happens if I call `cbp.icimvau(u32::MAX)`?

Some comments about the inputs of the other methods below.